web/app/js/components/TapEventTable.jsx

```diff
@@ -61,149 +85,86 @@ let tapColumns = (resourceType, filterOptions, ResourceLink) => [
-  let data = _.map(labels, (v, k) => {
-    return {
-      labelName: k,
-      labelVal: v
-    };
-  });
-  return (
-    <React.Fragment>
-      <div>{title}</div>
-      <Table
-        className="meta-table-nested"
-        columns={srcDstMetaColumns}
-        dataSource={data}
-        size="small"
-        rowKey={row => title.replace(" ", "_") + row.labelName + row.labelVal}
-        bordered={false}
-        showHeader={false}
-        pagination={false} />
-    </React.Fragment>
-  );
-};
+const requestInitSection = d => (
+  <React.Fragment>
+    <h3>Request Init</h3>
+    <Row gutter={8} className="expand-section-header">
+      <Col span={8}>Authority</Col>
+      <Col span={9}>Path</Col>
+      <Col span={2}>Scheme</Col>
+      <Col span={2}>Method</Col>
+      <Col span={3}>TLS</Col>
+    </Row>
+    <Row gutter={8}>
+      <Col span={8}>{_.get(d, "requestInit.http.requestInit.authority")}</Col>
+      <Col span={9}>{_.get(d, "requestInit.http.requestInit.path")}</Col>
+      <Col span={2}>{_.get(d, "requestInit.http.requestInit.scheme.registered")}</Col>
+      <Col span={2}>{_.get(d, "requestInit.http.requestInit.method.registered")}</Col>
+      <Col span={3}>{_.get(d, "base.tls")}</Col>
+    </Row>
+  </React.Fragment>
+);
+
+const responseInitSection = d => (
+  <React.Fragment>
+    <h3>Response Init</h3>
+    <Row gutter={8} className="expand-section-header">
+      <Col span={4}>HTTP Status</Col>
+      <Col span={4}>Latency</Col>
+    </Row>
+    <Row gutter={8}>
+      <Col span={4}>{_.get(d, "responseInit.http.responseInit.httpStatus")}</Col>
+      <Col span={4}>{formatTapLatency(_.get(d, "responseInit.http.responseInit.sinceRequestInit"))}</Col>
+    </Row>
+  </React.Fragment>
+);
+
+const responseEndSection = d => (
+  <React.Fragment>
+    <h3>Response End</h3>
+    <Row gutter={8} className="expand-section-header">
+      <Col span={4}>GRPC Status</Col>
+      <Col span={4}>Latency</Col>
+      <Col span={4}>Response Length (B)</Col>
+    </Row>
+    <Row gutter={8}>
+      <Col span={4}>{grpcStatusCodes[_.get(d, "responseEnd.http.responseEnd.eos.grpcStatusCode")]}</Col>
+      <Col span={4}>{formatTapLatency(_.get(d, "responseEnd.http.responseEnd.sinceResponseInit"))}</Col>
+      <Col span={4}>{formatWithComma(_.get(d, "responseEnd.http.responseEnd.responseBytes"))}</Col>
+    </Row>
+  </React.Fragment>
+);
 
 // hide verbose information
 const expandedRowRender = d => {
   return (
-    <Row gutter={8}>
-      <Col span={12}>{ renderMetaLabels("Source Metadata", _.get(d, "base.sourceMeta.labels", {})) }</Col>
-      <Col span={12}>{ renderMetaLabels("Destination Metadata", _.get(d, "base.destinationMeta.labels", {})) }</Col>
-    </Row>
+    <div className="tap-more-info">
+      <Row gutter={8} className="tap-info-section">
+        {requestInitSection(d)}
+      </Row>
+      <hr />
+      <Row gutter={8} className="tap-info-section">
+        {responseInitSection(d)}
```

Comment:
I'm not sure if this was a problem in the existing implementation or not, but I tried to expand a row for a request that had not finished yet, and I encountered the following JS error:

![image](https://user-images.githubusercontent.com/9226/45569211-af50d600-b813-11e8-949a-6780498d9a23.png)

I think the right fix here is to only add the response init and response end rows if `responseEnd.http.responseInit` and `responseEnd.http.responseEnd` are not null.

Reply:
oooh, good catch!